lib/Containers/MerkleTree.h

```diff
@@ -54,6 +54,51 @@ template <typename Hasher,
           std::uint64_t const BranchingBits = 3  // 8 children per internal node,
           >
 class MerkleTree {
+  // A MerkleTree has three parameters which define its semantics:
+  //  - rangeMin: lower bound (inclusive) for _rev values it can take
+  //  - rangeMax: upper bound (exclusive) for _rev values it can take
+  //  - depth: depth of the tree (root plus so many levels below)
+  // We call rangeMin-rangeMax the "width" of the tree.
+  //
+  // We do no longer grow trees in depth, since this would mean a complete
+  // rehash or a large growth of width, which can lead to integer overflow,
+  // which we must avoid.
+  //
+  // However, we do grow the width of a tree as needed. Originally, we
+  // only grew to the right and rangeMin was constant over the lifetime
+  // of the tree. This turned out to be not good enough, since we cannot
+  // estimate the smallest _rev value a collection will ever receive
+  // (for example, in DC2DC we might need to replicate old data into a
+  // newly created collection). Therefore, we can now also grow the width
+  // to the left by decreasing rangeMin.
+  //
+  // Unfortunately, we can only compare two different MerkleTrees, if
+  // the difference of their rangeMin values is a multiple of the number
+  // of _rev values in a leaf node, which is 
+  //   (rangeMax-rangeMin)/(BranchingFactor^depth),
```

Comment:
BranchingFactor is not used anymore, so anything related to it can actually be removed.

Reply:
BranchingFactor is currently hard-coded to 8. So we can remove it any time later from the code. It shouldn't change any user-visible parts of APIs.

Reply:
`BranchingFactor` is only stored and read from vpack slices, but otherwise not used _at all_, so I would simply remove it now (together with the static string).
In fact, I think this comment is no longer entirely accurate. @neunhoef can you comment on that?

Reply:
We still store the "depth" in the Meta information. It is currently a constant 6 everywhere. The number of buckets we use is defined to be `BranchingFactor^depth`, which is currently a constant `2^18`.
If we want to clean this up everywhere, we would have to:
 - remove `BranchingBits` and `BranchingFactor` in `MerkleTree.h`
 - change Meta and put `NumberOfBuckets` or so instead of `depth`
 - change all places which use depth
 - change all comments where depth and BranchingFactor occur
 - there are 33 occurrences of `nodeCountAtDepth` alone in the code, so this is not an insignificant change.

Reply:
Actually, I was wrong, in the unittests we do use MerkleTree objects with depth set to 2 and 3, in addition to the normal 6.

Reply:
The word BranchingFactor is removed from the code and from comments. We only use BranchingBits now to work with depth.